FrontBundle/SubQuery/SubQueryGeneratorManager.php

```diff
@@ -18,23 +18,40 @@ public function addStrategy(SubQueryGeneratorInterface $strategy)
     }
 
     /**
-     * @param array $blockParameters
+     * @param array $block
      * @param array $baseSubQuery
      *
      * @return array
      */
-    public function generate(array $blockParameters, array $baseSubQuery)
+    public function generate(array $block, array $baseSubQuery)
     {
-        $subQuery = $baseSubQuery;
-        /** @var SubQueryGeneratorInterface $strategy */
-        foreach ($this->strategies as $strategy) {
-            foreach ($blockParameters as $blockParameter) {
-                if ($strategy->support($blockParameter)) {
-                    $subQuery = array_merge($subQuery, $strategy->generate($blockParameter));
+        $subQuery = array_merge($baseSubQuery, $this->generateSubQueryCache($block));
+        if(isset($block['blockParameter'])) {
+            /** @var SubQueryGeneratorInterface $strategy */
+            foreach ($this->strategies as $strategy) {
+                foreach ($block['blockParameter'] as $blockParameter) {
+                    if ($strategy->support($blockParameter)) {
+                        $subQuery = array_merge($subQuery, $strategy->generate($blockParameter));
```

Comment:
@alavieille : it's not concerning all kind of strategies, but only those always run. If the manager checks only for the first supporting, this does not apply.
But ok let's do this